container.go

```diff
@@ -0,0 +1,342 @@
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// Shutdown requests a container shutdown, but it may not actually be shutdown until Wait() succeeds.
+func (container *hcsShimContainer) Shutdown() error {
+	title := "HCSShim::Container::Shutdown"
+	var (
+		options string
+		resultp *uint16
+	)
+
+	if err := hcsShutdownComputeSystem(container.handle, options, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// Terminate requests a container terminate, but it may not actually be terminated until Wait() succeeds.
+func (container *hcsShimContainer) Terminate() error {
+	title := "HCSShim::Container::Terminate"
+	var (
+		options string
+		resultp *uint16
+	)
+
+	if err := hcsTerminateComputeSystem(container.handle, options, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// Wait synchronously waits for the container to shutdown or terminate.
+func (container *hcsShimContainer) Wait() error {
+	_, err := container.waitTimeoutInternal(syscall.INFINITE)
+	return err
+}
+
+// WaitTimeout synchronously waits for the container to terminate or the duration to elapse. It
+// returns true if the event occurs successfully without timeout.
+func (container *hcsShimContainer) WaitTimeout(timeout time.Duration) (bool, error) {
+	// TODO can this overflow uint32
+	return container.waitTimeoutInternal(uint32(timeout.Nanoseconds() / (1000 * 1000)))
+}
+
+func (container *hcsShimContainer) waitTimeoutInternal(timeout uint32) (bool, error) {
+	title := "HCSShim::Container::waitTimeoutInternal"
+	var (
+		resultp   *uint16
+		exitEvent syscall.Handle
+	)
+
+	if err := hcsCreateComputeSystemWait(container.handle, &exitEvent, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return false, err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	s, e := syscall.WaitForSingleObject(syscall.Handle(exitEvent), timeout)
+	switch s {
+	case syscall.WAIT_OBJECT_0:
+		break
+	case syscall.WAIT_FAILED:
+		return false, makeError(e, title, "WaitForSingleObject failed")
+	default:
+		return false, errors.New("os: unexpected result from WaitForSingleObject")
+	}
+
+	return true, nil
+}
+
+func (container *hcsShimContainer) Properties() (*ContainerProperties, error) {
```

Comment:
@jhowardmsft @jstarks thoughts on this method? Should it return the whole ContainerProperties struct, or should that be internal and we only expose methods such as container.AreUpdatesPending(), container.IsRunning() etc. 

I'm leaning towards breaking this up and only exposing the methods that we chose rather than the struct from the HCS.